### PR TITLE
add func for stairlike links rendering

### PR DESCRIPTION
**Pull request by martyns0n**
I made stair-like link paths. I think it's missing functionality because it's useful for advanced hierarchy like organizational charts.

Examples:
Screenshot 2019-12-04 at 18 38 16

[VX-library](https://github.com/hshoff/vx) have a "step" type for links. You can configure and see it [here in action](https://vx-demo.now.sh/linkTypes)

### CODE REVIEW

File: src/Link/index.js [hunk 1]
```diff
@@ -32,6 +32,15 @@ export default class Link extends React.PureComponent {
     }
   }
 
+  drawStairPath(linkData, orientation) {
+    const { source, target } = linkData;
+    const deltaY = target.y - source.y;
+
+    return orientation === 'horizontal'
+      ? `M${source.x},${source.y} V${source.y + deltaY / 2} H${target.x} V${target.y}`
```
Comment (bkrem) [inline src/Link/index.js]: While test-driving your change on the demo I noticed that this ternary is the wrong way around, i.e. the nodes were rendering horizontally but the links vertically.

Flipping the ternary expression results in correct behaviour though so an easy fix :)

File: src/Link/index.js [hunk 2]
```diff
@@ -82,6 +91,10 @@ export default class Link extends React.PureComponent {
       return this.drawStraightPath(linkData, orientation);
     }
 
+    if (pathFunc === 'stair') {
```
Comment (bkrem) [inline src/Link/index.js]: Minor (grammar): When referring to the concept of stairs or a stair-like object in English, it's usually clearer to refer to the plural form (i.e. "stairs") if you want people to think of a stair-shaped graph. Kind of counter-intuitive I know.

Comment (bkrem) [reply src/Link/index.js]: I would leave `drawStairPath` untouched though. I'm only referring to the enum value for `pathFunc` :)

File: src/Link/tests/index.test.js [hunk 3]
```diff
@@ -111,6 +114,18 @@ describe('<Link />', () => {
     );
   });
 
+  it('return an appropriate stairPath accordin to `props.orientation`', () => {
```
Comment (bkrem) [inline src/Link/tests/index.test.js]: Typo: "according"

**Comment by bkrem** [comment]
Thank you for adding this path function + tests @martyns0n 💯 

Already looks great, just a few minor points to be resolved before we can ship this 👍

File: src/Link/index.js (hunk 1 above)
Comment (martyns0n) [reply src/Link/index.js]: Yeah, thanks. I'll fix it.

I have question how to run demo for fast visual changes. Can you describe it?

File: src/Link/index.js (hunk 2 above)
Comment (martyns0n) [reply src/Link/index.js]: I renamed it to "Step". It's more obvious i think.

File: src/Link/tests/index.test.js (hunk 3 above)
Comment (martyns0n) [reply src/Link/tests/index.test.js]: Fixed

**Comment by bkrem** [comment]
Published: https://github.com/bkrem/react-d3-tree/releases/tag/v1.16.0

Thanks again for taking the time to add this! :)

File: src/Link/index.js (hunk 1 above)
Comment (bkrem) [reply src/Link/index.js]: Yeah apologies if the demo is confusing; it's really not properly set up for public usage at the moment.

I've just updated the readme on the demo to simplify the process somewhat with `yarn/npm link`: https://github.com/bkrem/react-d3-tree-demo/

Basically, I symlink the library into the demo and have both running in watch mode (webpack on the lib, standard `nodemon` for the demo) to hot reload changes automatically.

Let me know if something is still unclear there or incorrect. Cleaning up the demo and making it more friendly for people to experiment with is one of the multiple issues I'm planning to address for this library when I'm on sabbatical early 2020.

**Pull request merged**